pkg/logs/status/status.go

```diff
@@ -1,38 +1,94 @@
 // Unless explicitly stated otherwise all files in this repository are licensed
 // under the Apache License Version 2.0.
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
-// Copyright 2017 Datadog, Inc.
+// Copyright 2018 Datadog, Inc.
 
 package status
 
+import (
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"strings"
+)
+
 var (
 	builder *Builder
 )
 
-// Source provides some information about a logs source
+// Source provides some information about a logs source.
 type Source struct {
-	Type   string `json:"type"`
-	Status string `json:"status"`
+	Type   string   `json:"type"`
+	Status string   `json:"status"`
+	Inputs []string `json:"inputs"`
+	// TCP, UDP
+	Port int `json:"port"`
+	// File
+	Path string `json:"path"`
+	// Docker
+	Image string `json:"image"`
+	Label string `json:"label"`
 }
 
-// Integration provides some information about a logs integration
+// Integration provides some information about a logs integration.
 type Integration struct {
 	Name    string   `json:"name"`
 	Sources []Source `json:"sources"`
 }
 
-// Status provides some information about logs-agent
+// Status provides some information about logs-agent.
 type Status struct {
 	IsRunning    bool          `json:"is_running"`
 	Integrations []Integration `json:"integrations"`
 }
 
-// Initialize instanciates builder
-func Initialize(sourcesToTrack []*SourceToTrack) {
-	builder = NewBuilder(sourcesToTrack)
+// Builder is used to build the status.
+type Builder struct {
```

Comment:
This `struct` only wraps `sources []*config.LogSource`, is it worth keeping it ?

Reply:
It's useful (to me) to pass around sources and abstract away how they are provided / filtered. Also I feel like manipulating a struct makes for easier refactoring than plain arrays. If this PR wasn't already too long I would probably have refactored all the places where we pass arrays of sources. I don't think it's worth worrying to much about it at the moment as this is likely going to change soon with auto config.

Reply:
👍